### ISSUE TITLE
Framebuffer not being copied?

### ISSUE THREAD

**Issue by m1se** [CplNathan/Nathans-Tarkov-Radar-Public#1]
Hello, 
I have been doing some basic testing with looking glass and can get the normal client working just fine no problems, but when using the modifed client it seems as if the framebuffer from the host is not being copied to the client resulting in the image below. 

![test](https://user-images.githubusercontent.com/48596874/80064961-1eada200-84ee-11ea-8393-819b7efc1d71.jpg)

**Comment by CplNathan** [comment]
Hi,
Unless something in the log says otherwise it is most likely something I added into the initialization function and not an issue with the framebuffer being copied although it probably causes similar symptoms.

I added three checks that would cause this issue on line 499, 511, and 517 of the egl.c in the egltarkov_render_startup function found in /renderers/EGLTarkov/egl.c although I can't see why these would fail unless you lacked enough memory and the arrays failed to allocate, anyhow they should show in the log if they are failing.

I really should add some on-screen debug information and improve the way the program handles the flow from opening the game to loading in-game as there are many places that the program will fail without clear reason if you don't run things in the correct order.

Anyway get back to me with anything suspicious in the terminal as there should be something.

**Comment by m1se** [comment]
I use my integrated gpu of my i9 9900ks, it seems to work with no hitches in the normal client of looking glass, I do not know if using an IGPU in this case would cause problems. When I launch the client after I started Tarkov all I get is.
`root@chiche:/home/cc/Downloads/Nathans-Tarkov-Radar-Public-looking-glass-edition/client/build# ./looking-glass-client -s -g EGLTarkov
1118419146 [I]               main.c:1564 | main                           | Looking Glass (B1-162-g3acfbdce02+1)
1118419159 [I]               main.c:1565 | main                           | Locking Method: Atomic
1118495669 [I]            ivshmem.c:180  | ivshmemOpenDev                 | KVMFR Device     : /dev/shm/looking-glass
1118495680 [I]               main.c:1248 | lg_run                         | Trying forced renderer
1118517927 [I]                egl.c:186  | egltarkov_initialize           | Double buffering is on
1118522341 [I]                egl.c:200  | egltarkov_initialize           | Multsampling enabled, max samples: 4

Initializing Game Offsets
Game:	Looping process 19566b000:EscapeFromTarkov.exe
	7ff9cedf0000	7ff9cedf1df0	b000	WINNSI.DLL
	7ff7abd30000	7ff7abd31268	a3000	EscapeFromTarkov.exe
	7ff9d7820000	00000000	1f0000	ntdll.dll
	7ff9d76a0000	7ff9d76b7c70	b2000	KERNEL32.DLL
	7ff9d4d80000	7ff9d4d8aa50	2a3000	KERNELBASE.dll
	7ff988dc0000	7ff989f25b18	16d5000	UnityPlayer.dll
Game Object Manager: 21d79cc4070	Local Game World: 0
1118595843 [I]               main.c:1060 | try_renderer                   | Using Renderer: EGLTarkov
1118661512 [I]               main.c:1394 | lg_run                         | Using Clipboard: X11
1118700888 [I]                egl.c:376  | egltarkov_render_startup       | Supported extensions: EGL_EXT_platform_base EGL_EXT_device_base EGL_EXT_device_enumeration EGL_EXT_device_query EGL_KHR_client_get_all_proc_addresses EGL_EXT_client_extensions EGL_KHR_debug EGL_KHR_platform_x11 EGL_EXT_platform_x11 EGL_EXT_platform_device EGL_EXT_platform_wayland EGL_MESA_platform_gbm EGL_MESA_platform_surfaceless
1118700899 [I]                egl.c:382  | egltarkov_render_startup       | use native: false
1118710158 [I]                egl.c:469  | egltarkov_render_startup       | Vendor  : Intel Open Source Technology Center
1118710168 [I]                egl.c:470  | egltarkov_render_startup       | Renderer: Mesa DRI Intel(R) UHD Graphics 630 (Coffeelake 3x8 GT2) 
1118710170 [I]                egl.c:471  | egltarkov_render_startup       | Version : OpenGL ES 3.2 Mesa 19.2.8
` 
Anyways thanks for the help.

**Comment by m1se** [comment]
So after some testing it seems like lgmpClientInit is returning LGMP_ERR_INVALID_SESSION or LGMP_ERR_INVALID_MAGIC continuously which is kinda weird. 
[https://github.com/CplNathan/Nathans-Tarkov-Radar-Public/blob/looking-glass-edition/client/src/main.c#L1434](url)

**Comment by CplNathan** [comment]
When you say that you can 'get the normal client to work' do you mean you have tried without the launch option `-g EGLTarkov` or are using a vesion from the master branch of looking-glass?

My thought is that you are using the latest version of looking-glass for your testing with the latest version of the windows client (and this works?) and I had not realized this, so if that is the case it seems likely that the version I forked has some differences with the master version of the windows client causing compatibility issues (I recall similar issues in my testing, not sure on the error).

Are you compiling the windows client from the master branch? I don't recall the exact version but try the binary version a12 from here (or compile it) https://github.com/gnif/LookingGlass/releases as it may actually be an issue with the frame-buffer after all.

**Comment by m1se** [comment]
Yes I actually just got it to work, turns out for the host I needed to download the latest "bleeding edge" build to get it to work. What I meant by getting the normal client to work was that I was able to run the master branch from looking glass repo. I just needed to read into some documentation after I found out that lgmpClientInit was not returning expected value. Once again thank you for your help and work!
